File: Reqnroll/Chapter18/18.4-SeededDataPattern/After/WIMP.Specs/Support/Hooks.cs
using Reqnroll;

using WIMP.App.Data;
using WIMP.Specs.Drivers;

namespace WIMP.Specs.Support;

[Binding]
public class Hooks(AppHostingContext appHostingContext, IDatabaseDriver databaseDriver, IDataRepository dataRepository, MenuBackdoorDriver menuBackdoorDriver)
{
    [BeforeScenario(Order = 0)]
    public async Task ResetDatabase()
    {
        await databaseDriver.UpgradeSchemaIfNeeded();
        await databaseDriver.EmptyDatabase();
        SeedMenuData();
    }

    private void SeedMenuData()
    {
        Console.WriteLine("seeding menu data");
        menuBackdoorDriver.SetMenuItems(
        [
            new MenuItemData { Name = "Margherita", Price = 7.99m, Calories = 900, Vegetarian = true },
            new MenuItemData { Name = "Pepperoni", Price = 9.99m, Calories = 1200 },
            new MenuItemData { Name = "Capricciosa", Price = 8.99m, Calories = 1100 }
        ]);
    }

    [BeforeScenario(Order = 1)]
    public void CreateAppHost()
    {
        appHostingContext.AppHost = new WimpAppHost(dataRepository);
    }

    [AfterScenario]
    public void DisposeAppHost()
    {
        appHostingContext.AppHost?.Dispose();
    }
}

File: CucumberJVM/Chapter21/21.1-TestEnvironmentInitializationPattern/Before/src/test/java/com/wimp/app/specs/support/Hooks.java
package com.wimp.app.specs.support;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import com.wimp.app.specs.drivers.DatabaseDriver;
import io.cucumber.java.*;

public class Hooks {
    protected static final Logger log = LoggerFactory.getLogger(Hooks.class);

    private final DatabaseDriver databaseDriver;

    public Hooks(DatabaseDriver databaseDriver) {
        this.databaseDriver = databaseDriver;
    }

    @Before(order = 0)
    public void resetDatabase() {
        databaseDriver.emptyDatabase();
    }
}
